src/mtcp/mtcp_restart.c

```diff
@@ -1262,10 +1273,32 @@ static void lock_file(int fd, char* name, short l_type)
   }
 }
 
+static char* fix_parent_directories(char* filename) {
+  int mtcp_sys_errno;
+  int i;
+  for (i=1; filename[i] != '\0'; i++) {
+    if (filename[i] == '/') {
+      int rc;
+      filename[i] = '\0';
+      rc = mtcp_sys_mkdir(filename, S_IRWXU);
+      if (rc < 0 && mtcp_sys_errno != EEXIST ) {
+        MTCP_PRINTF("error %d re-creating directory %s\n",
+		    mtcp_sys_errno, filename);
+        filename[i] = '/';  // Restore original filename
+	mtcp_abort();
+      } // else parent directory already exists or was re-created
+      // FIXME:  We should unlink parent directories if we re-created them.
+      filename[i] = '/';  // Restore original filename
+    }
+  }
+  return filename;  // Success.  All parent directories now exist.
```

Comment:
I'm primarily thinking of future extensions.  In one earlier variant, I had fix_parent_directories() returning NULL on failure, so that the parent function could do something further.  In a previous commit of mine to mtcp_restart.c, I had to do that treatment for mtcp_readfile(), changing it from succeed-or-abort to succeed-or-pass-error-to caller.  So, let's keep the potentially more general logic here.